### PR TITLE
11 add basic player upgrades AND 24 Implement Health and Bullet Damage System for Enemies

### PR DESCRIPTION
**Pull request by mitchellkazin15**
Major changes:
- Health component with methods to take damage and healing, and emit signal on change
- Exp component that can receive exp, and level up and emit signals on changes
- Enemies and players can now die
- Enemies give exp when killed by player
- Hud components now accurately track health and exp
- On level up, you get offered upgrades
    - Use arrow keys or d-pad to select upgrade direction
    - After selecting, you will get a new revolver facing the direction you chose
    
Minor changes:
- Made planet level larger ( 64x64 tiles)
- Added more enemies to level
- Fixed bug with enemies locking onto players header
- Fixed bug with camera inputs (thanks to Will)
- Moved player, enemy, bullets etc to _physics_process from _process for better physics consistency
- Code style clean up
- New button and health bar sprites

### CODE REVIEW

File: project.godot [hunk 1]
```diff
@@ -60,15 +60,6 @@ move_right={
...
 }
-
-[rendering]
-
-textures/canvas_textures/default_texture_filter=0
-zoom_in={
-"deadzone": 0.5,
-"events": [Object(InputEventMouseButton,"resource_local_to_scene":false,"resource_name":"","device":-1,"window_id":0,"alt_pressed":false,"shift_pressed":false,"ctrl_pressed":false,"meta_pressed":false,"button_mask":0,"position":Vector2(0, 0),"global_position":Vector2(0, 0),"factor":1.0,"button_index":4,"canceled":false,"pressed":false,"double_click":false,"script":null)
-]
-}
```
Comment (wchaney98) [inline project.godot:71]: I think this needs to be added back in, seeing lots of errors about this not existing

**Comment by wchaney98** [comment]
Scene itself looks good, should I be seeing the gun and be shooting the enemies? I'm just able to walk around but nothing is attacking the enemies.

**Comment by mitchellkazin15** [comment]
> Scene itself looks good, should I be seeing the gun and be shooting the enemies? I'm just able to walk around but nothing is attacking the enemies.

You gotta select an upgrade direction either with a d-pad or arrow keys and then you'll get a gun facing that direction

File: project.godot (hunk 1 above)
Comment (mitchellkazin15) [reply project.godot:71]: Oh yep good catch

**Review by wchaney98 — APPROVED** [review]

**Comment by wchaney98** [comment]
This is dope, it's a real game now! Thanks for doing all the extra cleanup too.

**Review by TheAdamSmith — APPROVED** [review]
Nice great changes! really transformed it into an actual game
Do we need the `.import` files to be tracked? this is something we maybe should have setup earlier but if they are not needed can we add it to the gitignore

**Pull request merged**